### DISCUSSION TITLE
RESTEasy Reactive - To block or not to block

### DISCUSSION THREAD

**Discussion by giscus** [quarkusio/quarkus#22555] [Quarkus Blog/Website]
# RESTEasy Reactive - To block or not to block

Quarkus: Supersonic Subatomic Java

https://quarkus.io/blog/resteasy-reactive-smart-dispatch/

**Comment by icedmac** [comment]
Such a very interesting topic !
question : does it mean that the only way to ensure a message as been effectively writed in a kafka topic is to mark the Rest method @transactional ?

**Reply by cescoffier** [reply]
No, the Kafka connector has an acknowledgment mechanism, unrelated to transaction (you also has Kafka transactions)

**Comment by fethier** [comment]
How is @transactional managing a distributed transaction across both kafka and hibernate data store? This feels like an example of the dual write problem or am I missing something.

**Reply by jgrabowski** [reply]
No it doesn't. In fact there's no possibility to wrap Kafka in XA transaction. I've already had this problem and one possible solution is to use Debezium there. Debezium is some kind of db change log to kafka proxy. It only pushes to Kafka things that were succesfully commited in the database.

**Reply by RomainWilbert** [reply]
@fethier or you could use the transactional outbox pattern
